### DISCUSSION TITLE
unable to customize button color

### DISCUSSION THREAD

**Discussion by ajayKomirishetty** [weareseeed/react-square-web-payments-sdk#52] [Help] [answered]
I have seen that we can customize input fields except Button color.

My requirement is I have a custom color and custom text to render on my button. I am looking for either of 2 options

1. ability to change iframe button color and text

            or

2.disable the iframe button and use my custom button.
can someone help me with this please? 

![image](https://user-images.githubusercontent.com/37865218/167742685-d9bb3da4-3a8d-4ae4-9de7-9ffcd56a9418.png)

**Comment by danestves** [comment, marked as answer]
Hi @ajayKomirishetty inside the `<CreditCardInput />` you have a `overrideStyles` property and there you can customize the styles of the button

**Reply by ajayKomirishetty** [reply]
Hey, I figured that out. It is working fine now. Thanks for the quick response. 

Is there any option to send params to disable button until user enters all the fields in the form.?

**Reply by danestves** [reply]
Unfortunately no, but you should not worry about that, since Square is the one that handles those validations, so if it is invalid it will not allow to submit the input

**Reply by ajayKomirishetty** [reply]
I am currently experiencing an error that says " card is not attached to window...". I have seen that there is an issue open for that and also I have noticed that this is fixed in version 3. Can you please let me know when are you planning to release v3? 

And also your documentation requires lot of changes. It took me a while to understand to dig in to the code and figure out how to pass params and how to use other functions.

**Reply by danestves** [reply]
The v3 will come out this week we a new documentation with all the aspect of props and configurations covered

**Reply by danestves** [reply]
v3 is out! https://github.com/weareseeed/react-square-web-payments-sdk/releases/tag/v3.0.0

**Reply by ajayKomirishetty** [reply]
another problem, I ofter experience is You never mentioned what are the necessary script, font and asset urls that we should whitelist on our side. I have added 10s of URLs to whitelist, but still I have these errors.
![image](https://user-images.githubusercontent.com/37865218/169535782-b47305cc-29dc-455c-816a-3eb5d140176e.png)

**Reply by danestves** [reply]
That is out of our scope, if you are using CSP rules you should add any of the URL's that Square is requesting, is not something with the wrapper
